### ISSUE TITLE
Show dialog prompt for Jenkins credentials

### ISSUE THREAD

**Issue by MarkusRohlof** [de-jcup/eclipse-jenkins-editor#76]
**Current Behaviour**
If a user does not explicitly set their User Credentials in `Window -> Preferences -> Jenins Editor -> User credentials...` any CLI call will fail with a somewhat undescriptive error message like so:
```
Jenkins CLI call failed.

JenkinsLinterCLIResult:
- exitCode:-2 (Linter was not executed at all)
- failureMessage:Access to Jenkins was not possible by command:

java -jar C:\path\to\jenkins-cli.jar -s https://jenkins.my-company.net:8181/ -http -auth :******* declarative-linter 

Maybe credentials not valid or hostname/firewall problems.
Please check Jenkins CLI setup in preferences
-Output was:
```

Also, in the settings view User credentials always appear to be set because the "password dots" are always visible so having to set your credentials is not the intuitive solution: 
![image](https://user-images.githubusercontent.com/5739492/91262483-5b292d00-e770-11ea-8ced-66e28eff9b02.png)

**Feature Request**
Please add a dialog prompt for when the user tries to access Jenkins via the Jenkins Linter or another function without having set their credentials first, prompting the user to insert their Jenkins credentials. 

A cherry on top would be to always ask for credentials when the validation failed since in Enterprise Companies you have to change your passwords somewhat regularly.

**Comment by de-jcup** [comment]
Hmm... let us clarify your wanted behaviour.

1. User did not set any credentials
start command (e.g linter) - check no credentials set, 
if not set : ==> show dialog for user credentials like in preferences + update  settings (or cancel if not wanted)
if set: ==> just start jenkins CLI command

2. CLI call failed (with Jenkins editor message containing "maybe credentials not valid or hostname/firewall problems.")
==> always show dialog for user credentials like in preferences+ update  settings (or cancel if not wanted)

I think this could be implemented easily - unfortunately Jenkins CLI jar does not give 100% reliability if the problem was a credential failure or firewall/hostname problem, or something else... The message shown is from Jenkins Editor by estimation... but I think showing up the dialog even when it's about other problems could be okay. 

@MarkusRohlof : Would this fit your needs?

**Comment by MarkusRohlof** [comment]
Yes, the behaviour as described would exactly fit my needs. 

Regarding 2b) a retry might be neat, but a text like `Jenkins CLI call failed. Please re-enter your credentials and try again.` would probably do just as well, since the user will not spend much time selecting the command again every once in a while. On the other hand, if you already implemented "password prompt -> call CLI" for 1. you might be able to just reuse that for 2, making the user experience more consistent. I'd say do it if  it does not result in ugly code 😄

**Comment by de-jcup** [comment]
I implemented a very simple approach- every command execution does now check if all necessary credential data is available for auth mode (currently only API-Token is avialble at the UI). The missing fields are shown in error message.

![image](https://user-images.githubusercontent.com/11644753/117214909-b8947400-adfd-11eb-9b2b-0b7447fb182f.png)

if only username is given, but api token missing the dialog text is different:
![image](https://user-images.githubusercontent.com/11644753/117215187-2e004480-adfe-11eb-8478-a70fec2f3c58.png)

**Comment by de-jcup** [comment]
@MarkusRohlof : This is a very simple approach and will be available with version 1.8.0 - please try out and create another issue if not suitable enough .
